### PR TITLE
Add objective as additional Module.

### PR DESCRIPTION
**Pull request by MarvinTeichmann**
Each model now is supposed to define an objective.
An objective should implement "decoder", "loss" and "evaluation".
An example implementation of softmax_classifier is given.

This commit also increases pep8 compatibility in touched files.

### CODE REVIEW

File: bin/tv-train [hunk 1]
```diff
@@ -158,13 +161,14 @@ def do_eval(hypes, eval_correct, phase, sess):
 
     # run evaluation on num_examples many images
     for step in xrange(steps_per_epoch):
-        start_time = time.time()
+        # start_time = time.time()
```
Comment (MartinThoma) [inline bin/tv-train]: This line should not only be commented out, but completely removed

File: bin/tv-train [hunk 2]
```diff
@@ -158,13 +161,14 @@ def do_eval(hypes, eval_correct, phase, sess):
...
         true_count += sess.run(eval_correct[phase])
-        duration = time.time() - start_time
+        # duration = time.time() - start_time
```
Comment (MartinThoma) [inline bin/tv-train]: This line should not only be commented out, but completely removed

File: examples/architectures/VGG8.py [hunk 3]
```diff
@@ -37,7 +35,8 @@ def _weight_variable(shape, stddev=0.01):
       stddev: standard deviation of a truncated Gaussian
     """
     initializer = tf.truncated_normal_initializer(stddev=stddev)
-    return tf.get_variable(name='weights', shape=shape, initializer=initializer)
+    return tf.get_variable(name='weights', shape=shape,
+                           initializer=initializer)
```
Comment (MartinThoma) [inline examples/architectures/VGG8.py:39]: This makes me happy :-) (And it shows another reason to have 80 character lines)

File: examples/architectures/VGG8.py [hunk 4]
```diff
@@ -115,8 +114,8 @@ def _fc_layer_with_dropout(bottom, name, size,
 
     with tf.variable_scope(name) as scope:
         n1 = bottom.get_shape()[1].value
-        stddev = (2/n1)**0.5
-        #stddev = 0.04
+        stddev = (2 / n1)**0.5
+        # stddev = 0.04
```
Comment (MartinThoma) [inline examples/architectures/VGG8.py]: This should be removed (or a comment should be added when one wants to have it - then a flag should be there to enable / disable it without changing the code)

File: examples/architectures/VGG8.py (hunk 3 above)
Comment (MarvinTeichmann) [inline examples/architectures/VGG8.py:39]: I do not quite understand the comment? What does make you happy?

**Comment by MartinThoma** [comment]
Could you please also adjust the README.md ("workflow"):
- Adjust the JSON-file under 'Create JSON file'
- Add the line "An objective should implement "decoder", "loss" and "evaluation"."

File: examples/architectures/VGG8.py (hunk 3 above)
Comment (MartinThoma) [inline examples/architectures/VGG8.py:39]: To see that our code gets more conformant to PEP8

**Comment by MarvinTeichmann** [comment]
I have added the suggestions. Should we merge the code now?

**Comment by MartinThoma** [comment]
If it works, go ahead. I can't have a look at it right now as my computers
fan makes stage noises.

**Comment by MarvinTeichmann** [comment]
It works.

**Comment by MartinThoma** [comment]
I am not sure how pull requests can be used nicely, but next time please merge the branch like this:

http://stackoverflow.com/a/36763755/562769 where `test` is the branch you want to merge into master.

The important point is `--squash`. It allows you to merge all commits of the feature-branch as a single commit into the master branch. You can also re-write the commit message. So in this case, your commit message should have been something like

```
Add objective as additional module

Each model now is supposed to define an objective.
An objective should implement "decoder", "loss" and "evaluation".
An example implementation of softmax_classifier is given.

This commit also increases pep8 compatibility in touched files and
removes outdated lines of code.
```

**Pull request merged**